### ISSUE TITLE
Sample extract build can result in merge conflicts committed to master

### ISSUE THREAD

**Issue by ewingjm** [ewingjm/development-hub#31]
The sample extract build does not check for the result of:

```powershell
  git merge origin/$SourceBranch --squash --no-commit;
```

This means that the extract will push even if the merge failed. We need to either prevent the merging of a solution merge that has merge conflicts with master or update the solution merge status to 'Merged' or 'Failed' dependent on the outcome of the extract build.

**Comment by ewingjm** [comment]
Using the following the script can force the extract script to fail - 

```powershell
if ($SourceBranch)
{
  git merge origin/$SourceBranch --squash --no-commit;
  $result = git merge HEAD
  if ($result[0] -like "*error*") 
  {
    Write-Error "Unable to automatically merge the source branch due to a conflict."
  }
}
```

Unfortunately, this only solves half of the problem. Either we need to stop further solution merges from happening or we need to fail an extract pipeline if a pipeline run if any previous runs have failed (suspect this is the easier option for the time being).

**Comment by ewingjm** [comment]
This would probably be resolved if we implemented my suggestion on #25.
